iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/transformation/dag/input/QueryDataSetInputLayer.java

```diff
@@ -20,309 +20,138 @@
 package org.apache.iotdb.db.queryengine.transformation.dag.input;
 
 import org.apache.iotdb.db.exception.query.QueryProcessException;
-import org.apache.iotdb.db.queryengine.transformation.api.LayerPointReader;
+import org.apache.iotdb.db.queryengine.transformation.api.LayerReader;
 import org.apache.iotdb.db.queryengine.transformation.api.YieldableState;
 import org.apache.iotdb.db.queryengine.transformation.dag.memory.SafetyLine;
 import org.apache.iotdb.db.queryengine.transformation.dag.memory.SafetyLine.SafetyPile;
-import org.apache.iotdb.db.queryengine.transformation.datastructure.row.ElasticSerializableRowRecordList;
+import org.apache.iotdb.db.queryengine.transformation.datastructure.iterator.RowListForwardIterator;
+import org.apache.iotdb.db.queryengine.transformation.datastructure.row.ElasticSerializableRowList;
 
+import org.apache.tsfile.block.column.Column;
 import org.apache.tsfile.enums.TSDataType;
-import org.apache.tsfile.utils.Binary;
-
-import java.io.IOException;
 
 public class QueryDataSetInputLayer {
 
-  private IUDFInputDataSet queryDataSet;
+  private TsBlockInputDataSet queryDataSet;
```

Comment:
Perhaps using interface IUDFInputDataSet is better?

Reply:
I think `TsBlockInputDataSet ` would provide more information than `IUDFInputDataSet `.

And `QueryDataSetInputLayer` must be `TsBlockInputDataSet`.